decoder/vaapidecoder_vp8.cpp

```diff
@@ -221,16 +221,14 @@ bool VaapiDecoderVP8::fillPictureParam(const PicturePtr&  picture)
     }
 
     for (i = 0; i < 4; i++) {
+        int8_t level;
         if (seg->segmentation_enabled) {
-            picParam->loop_filter_level[i] = seg->lf_update_value[i];
-            if (!seg->segment_feature_mode) {
-                picParam->loop_filter_level[i] +=
-                    m_frameHdr.loop_filter_level;
-                picParam->loop_filter_level[i] =
-                    CLAMP(picParam->loop_filter_level[i], 0, 63);
-            }
+            level = seg->lf_update_value[i];
+            if (!seg->segment_feature_mode)
+                level += m_frameHdr.loop_filter_level;
         } else
-            picParam->loop_filter_level[i] = m_frameHdr.loop_filter_level;
+            level = m_frameHdr.loop_filter_level;
```

Comment:
Hi,

+ add Author: Gwenole Beauchesne <gwenole.beauchesne@intel.com> to the
patch, from libgstvaapi commit f2b0d02.

While you are at it, please correct the vaapidecoder_vp8.c header to
include:
  Author: Gwenole Beauchesne <gwenole.beauchesne@intel.com>

after Halley's line. Thanks.

2015-07-14 9:35 GMT+02:00 Xu Guangxin <notifications@github.com>:

> In decoder/vaapidecoder_vp8.cpp
> <https://github.com/01org/libyami/pull/256#discussion_r34543695>:
>
> >          } else
> > -            picParam->loop_filter_level[i] = m_frameHdr.loop_filter_level;
> > +            level = m_frameHdr.loop_filter_level;
>
> better use { }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/01org/libyami/pull/256/files#r34543695>.
>



-- 
Gwenole Beauchesne
Intel Corporation SAS / 2 rue de Paris, 92196 Meudon Cedex, France
Registration Number (RCS): Nanterre B 302 456 199